modules/rostests/apitests/ntdll/UserModeException.c

```diff
@@ -0,0 +1,136 @@
+/*
+* PROJECT:         ReactOS api tests
+* LICENSE:         GPL - See COPYING in the top level directory
+* PURPOSE:         Test for stack overflow
+* PROGRAMMER:      Jérôme Gardou
+*/
+
+#include "precomp.h"
+#include <pseh/pseh2.h>
+
+typedef void (*PFUNC)(void);
+
+extern void Illegal_06(void);
+extern void Illegal_0F_0B(void);
+extern void Illegal_0F_38_0C(void);
+
+void Test_IllegalInstructionSingle(
+    PFUNC Func)
+{
+    PEXCEPTION_POINTERS ExcPtrs;
+    EXCEPTION_RECORD ExceptionRecord;
+    NTSTATUS Status = STATUS_SUCCESS;
+
+    RtlZeroMemory(&ExceptionRecord, sizeof(ExceptionRecord));
+
+    _SEH2_TRY
+    {
+        Func();
+    }
+    _SEH2_EXCEPT(ExcPtrs = GetExceptionInformation(),
+                 ExceptionRecord = *ExcPtrs->ExceptionRecord,
+                 EXCEPTION_EXECUTE_HANDLER)
+    {
+        Status = GetExceptionCode();
+    }
+    _SEH2_END;
+
+    ok_ntstatus(Status, STATUS_ILLEGAL_INSTRUCTION);
+    ok_long(ExceptionRecord.ExceptionCode, STATUS_ILLEGAL_INSTRUCTION);
+    ok_long(ExceptionRecord.ExceptionFlags, 0);
+    ok_ptr(ExceptionRecord.ExceptionRecord, NULL);
+    ok(((ULONG_PTR)ExceptionRecord.ExceptionAddress >= (ULONG_PTR)Test_IllegalInstructionSingle) &&
+       ((ULONG_PTR)ExceptionRecord.ExceptionAddress < 0x00007FFFFFFFFFFF),
```

Comment:
Should this be an equivalent constant for x86?

Reply:
I improved it to check for the actual address.